internal/metastore/kv/datacoord/kv_catalog.go

```diff
@@ -169,7 +169,10 @@ func (kc *Catalog) AlterSegment(ctx context.Context, newSegment *datapb.SegmentI
 	maps.Copy(kvs, segmentKvs)
 	if newSegment.State == commonpb.SegmentState_Flushed && oldSegment.State != commonpb.SegmentState_Flushed {
 		flushSegKey := buildFlushedSegmentPath(newSegment.GetCollectionID(), newSegment.GetPartitionID(), newSegment.GetID())
-		newSeg := &datapb.SegmentInfo{ID: newSegment.GetID()}
+		newSeg := &datapb.SegmentInfo{
+			ID:     newSegment.GetID(),
+			IsFake: newSegment.GetNumOfRows() == 0,
```

Comment:
The faked segment will only save into handleoff meta, It should not be saved here.

Reply:
`newSeg` is the `SegmentInfo` we write flush event, will not be saved. `newSegment` is which will be saved